core/lib/da_client/src/lib.rs

```diff
@@ -0,0 +1,32 @@
+pub mod types;
+
+use std::fmt;
+
+use async_trait::async_trait;
+use types::{DAError, DispatchResponse, InclusionData};
+
+/// Trait that defines the interface for the data availability layer clients.
+#[async_trait]
+pub trait DataAvailabilityClient: Sync + Send + fmt::Debug {
+    /// Dispatches a blob to the data availability layer.
+    async fn dispatch_blob(
+        &self,
+        batch_number: u32,
+        data: Vec<u8>,
+    ) -> Result<DispatchResponse, DAError>;
+
+    /// Fetches the inclusion data for a given blob_id.
+    async fn get_inclusion_data(&self, blob_id: String) -> Result<Option<InclusionData>, DAError>;
```

Comment:
Are there implementations benefitting from owning `blob_id`? IMO, `blob_id: &str` would be more idiomatic and would require less cloning in case of default DA clients.

Reply:
makes sense, changed it to `&str`